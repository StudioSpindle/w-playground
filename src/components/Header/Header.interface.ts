import { TypeLogo } from '../shared.types';

export interface HeaderProps {
  logo?: TypeLogo;
}
