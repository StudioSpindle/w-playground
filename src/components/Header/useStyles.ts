import { makeStyles, createStyles } from '@material-ui/styles';

const useStyles = makeStyles(() =>
  createStyles({
    logo: {
      marginBottom: '1rem',
      marginTop: '1rem'
    }
  })
);

export default useStyles;
