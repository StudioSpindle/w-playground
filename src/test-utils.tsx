import React from 'react';
import { render, RenderOptions, RenderResult, Queries } from '@testing-library/react';
import { MuiThemeProvider } from '@material-ui/core/styles';
import { MemoryRouter } from 'react-router-dom';
import themeSpec from './themes/default';
import { PlaygroundProvider } from './playground-context';
import '@testing-library/jest-dom/extend-expect';

const AllTheProviders: React.FunctionComponent<{}> = ({ children }) => (
  <PlaygroundProvider>
    <MuiThemeProvider theme={themeSpec.theme}>
      <MemoryRouter>{children}</MemoryRouter>
    </MuiThemeProvider>
  </PlaygroundProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions<Queries>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
