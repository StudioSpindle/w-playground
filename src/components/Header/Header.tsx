import React from 'react';
import { AppBar, Toolbar, Grid, Link, Typography, Box } from '@material-ui/core';
import { NavLink as RouterLink } from 'react-router-dom';
import useStyles from './useStyles';
import { usePlaygroundState } from '../../playground-context';
import { TypeLogo } from '../shared.types';

export interface HeaderProps {
  logo?: TypeLogo;
}

const Header: React.FC<HeaderProps> = ({ logo }) => {
  const classes = useStyles();
  const { isLoggedIn } = usePlaygroundState();

  return (
    <AppBar position="static" elevation={1}>
      <Toolbar>
        <Grid container justify="space-between" alignItems="center">
          <Grid item>
            <Grid container alignItems="center">
              <Grid item>
                <Link component={RouterLink} exact to="/">
                  <img className={classes.logo} src={logo} alt="Logo" />
                </Link>
              </Grid>
              <Grid item>
                <Box p={1}>
                  <Typography variant="body2" color="inherit">
                    (BETA)
                  </Typography>
                </Box>
              </Grid>
            </Grid>
          </Grid>
          {isLoggedIn && (
            <Grid className={classes.headerMenu} item>
              <Link className={classes.headerLink} component={RouterLink} to="/canvas">
                Schema Editor
              </Link>
              <Link className={classes.headerLink} component={RouterLink} to="/graphiql">
                GraphiQL Editor
              </Link>
            </Grid>
          )}
        </Grid>
      </Toolbar>
    </AppBar>
  );
};

export default Header;
