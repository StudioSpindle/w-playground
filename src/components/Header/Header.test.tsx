import React from 'react';
import { render } from '../../test-utils';
import Header from './Header';

it('matches snapshot', () => {
  const container = render(<Header />);
  expect(container).toMatchSnapshot();
});
