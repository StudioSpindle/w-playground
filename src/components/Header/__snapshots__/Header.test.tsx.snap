// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation1 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <a
                    aria-current="page"
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary active"
                    href="/"
                  >
                    <img
                      alt="Logo"
                      class="makeStyles-logo-1"
                    />
                  </a>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <div
                    class="MuiBox-root MuiBox-root-184"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorInherit"
                    >
                      (BETA)
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation1 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <a
                  aria-current="page"
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary active"
                  href="/"
                >
                  <img
                    alt="Logo"
                    class="makeStyles-logo-1"
                  />
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiBox-root MuiBox-root-184"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorInherit"
                  >
                    (BETA)
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
